test/cmdlineargs.jl

```diff
@@ -597,6 +597,34 @@ let exename = `$(Base.julia_cmd()) --startup-file=no --color=no`
             "Int(Base.JLOptions().fast_math)"`)) == JL_OPTIONS_FAST_MATH_DEFAULT
     end
 
+    # --pkgimages with multiple cpu targets
+    @testset let v = readchomperrors(`$exename
+        --cpu-target='generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)'
+        --pkgimages=no`)
+
+        @test !v[1]
+        @test isempty(v[2])
+        @test v[3] == "ERROR: More than one command line CPU targets specified without a `--output-` flag specified"
+    end
+
+    @test success(`$exename
+        --cpu-target='generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)'
+        --pkgimages=yes`)
+
+
+    # Sysimage with multiple cpu targets
+    @testset "Sysimage for multiple microarchitecture" begin
+        sysimage= unsafe_string(Base.JLOptions().image_file)
+        outputo_file = tempname()
+        write(outputo_file, "1")
+        object_file = tempname() * ".o"
+        @test success(`$exename
+            --sysimage=$sysimage
+            --cpu-target='generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)'
+            --output-o=$object_file $outputo_file
+            --pkgimages=no`)
+    end
```

Comment:
this test sounds slow, and should not be included without further reconsideration of performance

Reply:
Yes. I've noticed that this test adds ~160s. It is too much time for 1 single test.
So, for easy dropping, I've added it in a dedicated commit.

I couldn't think of any other simpler test for output-o, which is faster, though.

Reply:
@vtjnash How about this: we intentionally don't pass the sysimage arg. When it fails we know everything before that passed, including our case. This takes only 0.05s

```
# Sysimage with multiple cpu targets
@testset "Sysimage for multiple microarchitectures" begin
    julia_path = joinpath(Sys.BINDIR, Base.julia_exename())
    outputo_file = tempname()
    write(outputo_file, "1")
    object_file = tempname() * ".o"

    # This is to test that even with `pkgimages=no`, we can create object file
    # with multiple cpu-targets
    # The cmd is checked for --object-o as soon as it is run. So, to avoid long
    # testing times, intentionally don't pass sysimage; when we reach the
    # corresponding error, we know that `check_cmdline` has already passed
    let v = readchomperrors(`$julia_path
        --cpu-target='native;generic'
        --output-o=$object_file $outputo_file
        --pkgimages=no`)

        @test v[1] == false
        @test v[2] == ""

        err = first(split(v[3], "\n\n"))
        @test err != "fatal: error thrown and no exception handler available.\nErrorException(\"More than one command line CPU targets specified without a `--output-` flag specified\")"
        @test err == "ERROR: File \"boot.jl\" not found"
    end
end
```

Reply:
yes, though let's do `@test !contains(err, "More than one command line CPU targets specified")` so it will still work if other minor aspects change (though the second test for `==` will also catch those)